app/src/main/java/ch/epfl/sdp/appart/utils/serializers/AdSerializer.java

```diff
@@ -0,0 +1,36 @@
+package ch.epfl.sdp.appart.utils.serializers;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import ch.epfl.sdp.appart.ad.Ad;
+import ch.epfl.sdp.appart.ad.PricePeriod;
+import ch.epfl.sdp.appart.database.firebaselayout.AdLayout;
+
+public class AdSerializer implements Serializer<Ad> {
+    @Override
+    public Map<String, Object> serialize(Ad data) {
+        Map<String, Object> adData = new HashMap<>();
+        adData.put(AdLayout.ADVERTISER_ID, data.getAdvertiserId());
+        adData.put(AdLayout.CITY, data.getCity());
+        adData.put(AdLayout.DESCRIPTION, data.getDescription());
+        adData.put(AdLayout.VR_TOUR, data.hasVRTour());
+        adData.put(AdLayout.PRICE, data.getPrice());
+        adData.put(AdLayout.PRICE_PERIOD, data.getPricePeriod().ordinal());
+        adData.put(AdLayout.STREET, data.getStreet());
+        adData.put(AdLayout.TITLE, data.getTitle());
+        return adData;
+    }
+
+    /**
+     * This method has not been implemented, because the photo references of an ad are not serialized with the rest of the data.
+     * Thus, it is not possible to simply "unbox" a map to an ad just like that.
+     * @param id
+     * @param serializedData
+     * @return
+     */
+    @Override
+    public Ad deserialize(String id, Map<String, Object> serializedData) {
+        return null;

```

Comment:
The deserialization of the ad is still in the Database class ?

Reply:
More or less : an ad is not really serialized in one part: first it must upload all the pictures of the ad, and then the ad's datas (poster's id, street, price, etc ...). As it is done in multiple parts, an AdBuilder class has been created (I don't know who created it) which essentially does the same thing. So the deserialization part is not really done in the Database but with the AdBuilder, which makes more sens than doing it with a deserializer.